python/ray/util/client/server/proxier.py

```diff
@@ -608,6 +608,23 @@ def prepare_runtime_init_req(
     return (init_request, new_job_config)
 
 
+class RequestIteratorProxy:
+    def __init__(self, request_iterator):
+        self.request_iterator = request_iterator
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            return next(self.request_iterator)
+        except grpc.RpcError:
+            # stop iteration if the stream is CANCLLED
```

Comment:
Can you add `logger.exception("Iterator failed with the following exception:")` here, that the cause of error can be logged somewhere.

Reply:
Hi @ckw017, Instead of using `logger.exception`, I use `logger.warning` instead. Since `logger.exception` is not very helpful in the case of `grpc.RpcError`. Please see the below screenshot.